packages/vector_graphics_codec/lib/vector_graphics_codec.dart

```diff
@@ -0,0 +1,631 @@
+  /// The codec is stateless and the const constructor should be preferred.
+  const VectorGraphicsCodec();
+
+  static const int _pathTag = 27;
+  static const int _fillPaintTag = 28;
+  static const int _strokePaintTag = 29;
+  static const int _drawPathTag = 30;
+  static const int _drawVerticesTag = 31;
+  static const int _moveToTag = 32;
+  static const int _lineToTag = 33;
+  static const int _arcToTag = 34;
+  static const int _closeTag = 35;
+
+  static const int _version = 1;
+  static const int _magicNumber = 0x00882d62;
+
+  /// Decode the vector_graphics binary.
+  ///
+  /// Without a provided [VectorGraphicsCodecListener], this method will only
+  /// validate the basic structure of an object. decoders that wish to construct
+  /// a dart:ui Picture object should implement [VectorGraphicsCodecListener].
+  ///
+  /// Throws a [StateError] If the message is invalid.
+  void decode(ByteData data, VectorGraphicsCodecListener? listener) {
+    final _ReadBuffer buffer = _ReadBuffer(data);
+    if (data.lengthInBytes < 5) {
+      throw StateError('The provided data was not a vector_graphics binary asset.');
+    }
+    final int magicNumber = buffer.getUint32();
+    if (magicNumber != _magicNumber) {
+      throw StateError('The provided data was not a vector_graphics binary asset.');
+    }
+    final int version = buffer.getUint8();
+    if (version != _version) {
+      throw StateError('The provided data does not match the currently supported version.');
+    }
+    while (buffer.hasRemaining) {
+      final int type = buffer.getUint8();
+      switch (type) {
+        case _fillPaintTag:
+          return _readFillPaint(buffer, listener);
+        case _strokePaintTag:
+          return _readStrokePaint(buffer, listener);
+        case _pathTag:
+          return _readPath(buffer, listener);
+        case _drawPathTag:
+          return _readDrawPath(buffer, listener);
+        case _drawVerticesTag:
+          return _readDrawVertices(buffer, listener);
+        case _moveToTag:
+          return _readMoveTo(buffer, listener);
+        case _lineToTag:
+          return _readLineTo(buffer, listener);
+        case _arcToTag:
+          return _readArcTo(buffer, listener);
+        case _closeTag:
+          return _readClose(buffer, listener);
+        default:
+          throw StateError('Unknown type tag $type');
+      }
+    }
+  }
+
+  /// Encode a draw path command in the current buffer.
+  ///
+  /// Requires that [pathId] and [paintId] to already be encoded.
+  void writeDrawPath(
+    VectorGraphicsBuffer buffer,
+    int pathId,
+    int paintId,
+  ) {
+    buffer._putUint8(_drawPathTag);
+    buffer._putInt32(pathId);
+    buffer._putInt32(paintId);
+  }
+
+  /// Encode a draw vertices command in the current buffer.
+  ///
+  /// The [indexes] are the index buffer used and is optional.
+  void writeDrawVertices(
+    VectorGraphicsBuffer buffer,
+    Float32List vertices,
+    Uint16List? indexes,
+    int? paintId,
```

Comment:
Are we punting on colors for now?

Reply:
We can add it later - I think we should use paints for now right?

Reply:
Sgtm